### DISCUSSION TITLE
Changing file_template option doesn't seem to have any effects

### DISCUSSION THREAD

**Discussion by Timelessprod** [miguelgrinberg/Flask-Migrate#549] [Q&A] [answered]
I am working on a Flask app relying on Flask-Sqlalchemy and Flask-Migrate.

I set up some configs for the project but it seems Flask-Migrate is ignoring the `file_template` option of the `[alembic]` section of the `alembic.ini` file. I have other options set up (`script_location`, `prepend_sys_path`, etc.) which work well, so I'm sure the config file is well parsed by Alembic.

Indeed instead of using the default file template `%%(rev)s_%%(slug)s` I want to use this one `%%(year)d_%%(month).2d_%%(day).2d_%%(rev)s_%%(slug)s` so my version files are automatically ordered chronologically and I can debug things easier when needing to go through migration changes.

So currently when running `flask db revision --autogenerate -m "My Message"` I get a new file called `eb18f808c6b7_my_message.py` while I expect `2024_07_22_eb18f808c6b7_my_message.py`

From what I read online, my template is correct and is supposed to work fine with Alembic, so why isn't it working here ? Is there anything specific to do compared to regular alembic ?

Thank you very much!

**Comment by miguelgrinberg** [comment, marked as answer]
It works fine for me. Did you notice that alembic.ini is located inside the `migrations` directory when using Flask-Migrate? That is where you need to add your custom settings.

**Reply by Timelessprod** [reply]
Indeed it fixed the problem. I had the alembic file aboe migrations (project architecture legacy) and specified the path to the alembic.ini in the env.py to the directory above migrations and it loaded everything well but this one option. Thanks for the hint!
